tower/src/spawn_ready/service.rs

```diff
@@ -1,59 +1,57 @@
-use super::future::{background_ready, ResponseFuture};
+use super::future::ResponseFuture;
+use crate::{util::ServiceExt, BoxError};
 use futures_core::ready;
 use futures_util::future::TryFutureExt;
 use std::{
     future::Future,
     pin::Pin,
     task::{Context, Poll},
 };
-use tokio::sync::oneshot;
 use tower_service::Service;
 
 /// Spawns tasks to drive an inner service to readiness.
 ///
 /// See crate level documentation for more details.
 #[derive(Debug)]
-pub struct SpawnReady<T> {
-    inner: Inner<T>,
+pub struct SpawnReady<S> {
+    inner: Inner<S>,
 }
 
 #[derive(Debug)]
-enum Inner<T> {
-    Service(Option<T>),
-    Future(oneshot::Receiver<Result<T, crate::BoxError>>),
+enum Inner<S> {
+    Service(Option<S>),
+    Future(tokio::task::JoinHandle<Result<S, BoxError>>),
 }
 
-impl<T> SpawnReady<T> {
+impl<S> SpawnReady<S> {
     /// Creates a new [`SpawnReady`] wrapping `service`.
-    pub fn new(service: T) -> Self {
+    pub fn new(service: S) -> Self {
         Self {
             inner: Inner::Service(Some(service)),
         }
     }
 }
 
-impl<T, Request> Service<Request> for SpawnReady<T>
+impl<S, Req> Service<Req> for SpawnReady<S>
 where
-    T: Service<Request> + Send + 'static,
-    T::Error: Into<crate::BoxError>,
-    Request: Send + 'static,
+    Req: 'static,
+    S: Service<Req> + Send + 'static,
+    S::Error: Into<BoxError> + 'static,
```

Comment:
Is this additional lifetime a breaking change?

Reply:
Turns out it's not actually needed; removed.